File: Source/trunk/SqlServerStorage/Models/ColumnSetting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sioux.TaskManagement.Models
{
    [Table("ColumnSettings")]
    public class ColumnSetting
    {
		[Key]
		public Guid Id { get; set; }

		[Required]
		public Guid ColumnId { get; set; }

		[Required]
		public Guid Next { get; set; }

		[Required]
		public Guid BoardId { get; set; }

		[Required]
		public DateTime Created { get; set; }

		[Required]
		public DateTime Updated { get; set; }

		[Required]
		public Status Status { get; set; }

		//[ForeignKey("ColumnId")]
		//public Column CurrentColumn { get; set; }

		//[ForeignKey("Next")]
		//public Column NextColumn { get; set; }

		public static implicit operator ColumnSettingInfo(ColumnSetting columnSetting)
		{
			return new ColumnSettingInfo()
			{
				Id = columnSetting.Id,
				ColumnId = columnSetting.ColumnId,
				Next = columnSetting.Next,
				BoardId = columnSetting.BoardId,
				Created = columnSetting.Created,
				Updated = columnSetting.Updated,
				Status = columnSetting.Status
			};
		}
    }
}

File: Source/trunk/WebApplication/WebApplication/webapp/js/ColumnSetting.js
var ColumnSetting = function (params) {
	this.id = null;
	this.columnId = null;
	this.next = 0;
	this.boardId = null;
	this.created = new Date();
	this.updated = null;
	this.status = 1;

	if (typeof params === 'string') {
		LOG && log('params === string');
		ColumnSetting.getById(params, function (columnSetting) {
			this.id = columnSetting.id;
		});

	} else if (typeof params === 'object') {
		LOG && log('params === object');
		this.id = params.id;
		this.columnId = params.columnId;
		this.next = params.next;
		this.boardId = params.boardId;
		this.created = params.created;
		this.updated = params.updated;
		this.status = params.status;
	} else {
		this.id = generateId('column-setting');
	}
}